### PR TITLE
Applies #690 - find_homography_dlt performance improvement and weight…

### PR DESCRIPTION
**Pull request by dkoguciuk**
…s params made optional

### Description

3 changes:
1) weights argument of find_homography_dlt is now optional
2) the whole function is made a bit more efficient (get rid of for loop)
3) Info printed if SVD fails

### Status
Ready

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/geometry/homography.py [hunk 1]
```diff
@@ -35,24 +36,22 @@ def find_homography_dlt(points1: torch.Tensor, points2: torch.Tensor, weights: t
...
-    w = torch.stack(w_list, dim=1)
-
-    # apply weights
-    w_diag = torch.diag_embed(w)
-    A = A.transpose(-2, -1) @ w_diag @ A
     try:
         U, S, V = torch.svd(A)
     except:
-        return torch.empty(points1_norm.size(0), 3, 3)
+        print('SVD did not converged')
```
Comment (ducha-aiki) [inline kornia/geometry/homography.py:53]: `RuntimeWarning('SVD did not converged')
`

File: kornia/geometry/homography.py [hunk 2]
```diff
@@ -7,15 +7,16 @@
 TupleTensor = Tuple[torch.Tensor, torch.Tensor]
 
 
-def find_homography_dlt(points1: torch.Tensor, points2: torch.Tensor, weights: torch.Tensor) -> torch.Tensor:
+def find_homography_dlt(points1: torch.Tensor, points2: torch.Tensor, weights: torch.Tensor = None) -> torch.Tensor:
```
Comment (ducha-aiki) [inline kornia/geometry/homography.py:10]:
```
from typing import Optional 

...
weights: Optional[ torch.Tensor]
```

**Review by ducha-aiki — CHANGES_REQUESTED** [review]
LGTM except things I commented :)

File: kornia/geometry/homography.py (hunk 1 above)
Comment (dkoguciuk) [reply kornia/geometry/homography.py]: Git it!

File: kornia/geometry/homography.py (hunk 2 above)
Comment (dkoguciuk) [reply kornia/geometry/homography.py]: Got it!

**Comment by dkoguciuk** [comment]
@ducha-aiki , fixed.

File: kornia/geometry/homography.py [hunk 3]
```diff
@@ -1,21 +1,26 @@
+import warnings
+
 import torch
 from typing import Tuple
+from typing import Optional
```
Comment (edgarriba) [inline kornia/geometry/homography.py:5]: merge both lines

File: kornia/geometry/homography.py [hunk 4]
```diff
@@ -1,21 +1,26 @@
...
 
 import kornia
 from kornia.geometry.epipolar import normalize_points
 
 TupleTensor = Tuple[torch.Tensor, torch.Tensor]
 
 
-def find_homography_dlt(points1: torch.Tensor, points2: torch.Tensor, weights: torch.Tensor) -> torch.Tensor:
+def find_homography_dlt(points1: torch.Tensor, points2: torch.Tensor, weights: Optional[torch.Tensor] = None) -> \
```
Comment (edgarriba) [inline kornia/geometry/homography.py:13]: remove backslash and reformat as:

```python
def find_homography_dlt(
    points1: torch.Tensor, points2: torch.Tensor, weights: Optional[torch.Tensor] = None) -> torch.Tensor
```

or

```python
def find_homography_dlt(
    points1: torch.Tensor, points2: torch.Tensor, weights: Optional[torch.Tensor] = None
) -> torch.Tensor
```

or use: `make autopep8` from project root

File: kornia/geometry/homography.py [hunk 5]
```diff
@@ -35,24 +40,22 @@ def find_homography_dlt(points1: torch.Tensor, points2: torch.Tensor, weights: t
...
-
-    # apply weights
-    w_diag = torch.diag_embed(w)
-    A = A.transpose(-2, -1) @ w_diag @ A
     try:
         U, S, V = torch.svd(A)
     except:
-        return torch.empty(points1_norm.size(0), 3, 3)
+        warnings.warn('SVD did not converge', RuntimeWarning)
+        return torch.empty((points1_norm.size(0), 3, 3), device=points1.device)
```
Comment (edgarriba) [inline kornia/geometry/homography.py:58]: add empty line (optional)

**Review by edgarriba — CHANGES_REQUESTED** [review]
minor changes mostly for format. Could you add a small test in here verifying when weights are None ?
https://github.com/kornia/kornia/blob/master/test/geometry/test_homography.py#L12

File: kornia/geometry/homography.py (hunk 3 above)
Comment (dkoguciuk) [reply kornia/geometry/homography.py:5]: Done.

File: kornia/geometry/homography.py (hunk 4 above)
Comment (dkoguciuk) [reply kornia/geometry/homography.py:13]: Done

File: kornia/geometry/homography.py (hunk 5 above)
Comment (dkoguciuk) [reply kornia/geometry/homography.py:58]: Done

**Review by edgarriba — APPROVED** [review]
LGTM

**Comment by dkoguciuk** [comment]
@ducha-aiki, we still need your approval to merge I believe :)

**Review by ducha-aiki — APPROVED** [review]
